ui/src/message_popup/author/new_question_page.jsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+
+import QuestionPage from './question_page.jsx';
+import {allArchivedQuestions} from './archived_questions.js';
+import {allQuestions} from '../questions.js';
+import * as Api from '../../helpers/api.js';
+
+
+export default React.createClass({
+  displayName: 'NewQuestionPage',
+
+  getInitialState(){
+    return ({
+      loaded: false,
+      allDatabaseQuestions: {
+        currentQuestions: allQuestions,
+        archivedQuestions: allArchivedQuestions
+      }
+    });
+  },
+
+  componentDidMount(){
+    Api.questionsQuery().end(this.onQuestionsReceived);
+  },
+
+  onQuestionsReceived(err, response){
+    const questions = JSON.parse(response.text).questions;
+    if(questions !== undefined){
```

Comment:
Ok, I now see why I `err` was `null`. If it is in development mode it returned `response.json({})` which caused `JSON.parse(response.text).questions` to be `undefined`. It shouldn't happen in the actual build so I'll change it to search for `err` instead of checking for `undefined`. Also, if in development mode (mainly because a failed get query to the database crashes my localhost, I'm not too sure if this is universal) I think we should just return an empty questions object structured to be like the actual one:
```
response.json({questions: {currentQuestions: [], archivedQuestion: []}})
```